### ISSUE TITLE
Environment variables are not honored by ./build.sh 

### ISSUE THREAD

**Issue by platipodium** [noaa-ocs-modeling/CoastalApp#126]
env variables to third party libs are not honored

```
$env |grep METIS

PARMETISHOME=/sciclone/home20/clemmen/devel/schism/schism/src/ParMetis-4.0.3/build/Linux-x86_64
METISHOME=/sciclone/home20/clemmen/devel/schism/schism/src/ParMetis-4.0.3/build/Linux-x86_64

$ ./build.sh -tp "metis parmetis"

   METISHOME      = /sciclone/home20/clemmen/devel/noaa/CoastalApp/THIRDPARTY_INSTALL
    PARMETISHOME   = /sciclone/home20/clemmen/devel/noaa/CoastalApp/THIRDPARTY_INSTALL
```

**Comment by pvelissariou1** [comment]
The third party components that are supported are datetime and parmetis.
the script scripts/download_parmetis.sh downloads parmetis, metis and GKlib into
thirdparty_open/parmetis.
The script and the models (WW3) look into $PARMETISHOME/lib and $PARMETISHOME/include to find the relevant files.

**Comment by platipodium** [comment]
> The third party components that are supported are datetime and parmetis.
> the script scripts/download_parmetis.sh downloads parmetis, metis and GKlib into
> thirdparty_open/parmetis.
> The script and the models (WW3) look into $PARMETISHOME/lib and $PARMETISHOME/include to find the relevant files.

Does not work for me, see #127

**Comment by platipodium** [comment]
I got it to PARMETIS env variables to work by updating everything.   There is still a minor issue with  `ESMFMKFILE` environment variable @pvelissariou1 

```
$> env |grep ESMFMKFILE
ESMFMKFILE=/usr/local/skylake/esmf-8.2.0-openmpi/lib/libO/Linux.intel.64.openmpi.default/esmf.mk
```

```
$> n./build.sh -plat femto -component "SCHISM"  -compiler intel
...

    ESMFMKFILE     = UNDEF

```

And subsequently

```
make[1]: Entering directory `/sciclone/home20/clemmen/devel/noaa/CoastalApp/SCHISM/schism-esmf'
src/include/Rules.mk:142: *** ESMF Makefile snippet variable not defined.  Schluss.
```

**Comment by pvelissariou1** [comment]
Hi Carsten,

Is esmf loaded as a module? If yes this is the issue when running the build
script for femto (envmodules_intel.femto):
*module purge <--*
module load intel/2018
module load openmpi/3.1.4/intel-2018
module load netcdf/4.4.1.1/intel-2018
module load netcdf-fortran/4.4.4/intel-2018
module load cmake/3.14.4

"module purge" removes all previously loaded modules and then the script
loads from scratch
all required modules. You are missing:
module load esmf
or something similar.

Takis

Panagiotis Velissariou, Ph.D., P.E.
UCAR Scientist
National Ocean and Atmospheric Administration
National Ocean Service
Office of Coast Survey CSDL/CMMB
Project Lead - Coastal Coupling
cell: (205) 227-9141
email: ***@***.***

**Comment by platipodium** [comment]
@pvelissariou1 I will look into this.  There might be an interference between a manually set ESMFMKFILE in the environment and the `module` commands.

**Comment by platipodium** [comment]
I fixed this in 38c3e0f0b9bd3de443b67721166b8fe2ae247a66
